File: testsuite/Jogl2Examples/PolyhedralBoundedSolidExample/src/DebuggerMouseInteractionTechniques.java
import java.awt.event.MouseEvent;
import models.DebuggerModel;
import java.awt.event.MouseWheelEvent;

import vsdk.toolkit.gui.AwtSystem;

public class DebuggerMouseInteractionTechniques
{
    public void processMouseEntered(DebuggerModel model)
    {
        if ( model.getCanvas() != null ) {
            model.getCanvas().requestFocusInWindow();
        }
    }

    public void processMouseExited(DebuggerModel model)
    {
        ;
    }

    public boolean processMousePressed(DebuggerModel model, MouseEvent e)
    {
        return model.getCameraController().processMousePressedEvent(AwtSystem.awt2vsdkEvent(e));
    }

    public boolean processMouseReleased(DebuggerModel model, MouseEvent e)
    {
        return model.getCameraController().processMouseReleasedEvent(AwtSystem.awt2vsdkEvent(e));
    }

    public boolean processMouseClicked(DebuggerModel model, MouseEvent e)
    {
        return model.getCameraController().processMouseClickedEvent(AwtSystem.awt2vsdkEvent(e));
    }

    public boolean processMouseMoved(DebuggerModel model, MouseEvent e)
    {
        return model.getCameraController().processMouseMovedEvent(AwtSystem.awt2vsdkEvent(e));
    }

    public boolean processMouseDragged(DebuggerModel model, MouseEvent e)
    {
        return model.getCameraController().processMouseDraggedEvent(AwtSystem.awt2vsdkEvent(e));
    }

    public boolean processMouseWheelMoved(DebuggerModel model, MouseWheelEvent e)
    {
        System.out.println(".");
        return model.getCameraController().processMouseWheelEvent(AwtSystem.awt2vsdkEvent(e));
    }
}

File: typescript/testsuite/VitralWebTestsuiteContainer/src/WebGLExamples/PolyhedralBoundedSolidExample/gui/debugger-mouse-interaction-techniques.ts
import type { DebuggerModel } from '../models/debugger-model';

/**
 * Port of
 * `java/testsuite/Jogl4Examples/PolyhedralBoundedSolidExample/src/gui/DebuggerMouseInteractionTechniques.java`.
 *
 * Every technique forwards to the model's camera controller and answers
 * whether the view has to be repainted, as Java's do; the wheel handler prints
 * its `.` first, as Java's does. The controller is the container's stand-in
 * for `CameraControllerOrbiter`.
 *
 * Java's `processMouseEntered` asks the canvas for the keyboard focus and its
 * `processMouseExited` does nothing; a browser element takes focus on a
 * pointer press, which is where the component asks for it. Java's clicked and
 * moved handlers forward to controller handlers the orbiter answers `false`
 * to, and a page delivers pointer `down`, `move` and `up`, so those collapse
 * into the three below — the reading the `PolygonClippingExample` module
 * records.
 */
export class DebuggerMouseInteractionTechniques {
  processMousePressed(model: DebuggerModel, e: PointerEvent, canvas: HTMLCanvasElement): boolean {
    return model.getCameraController().processPointerDown(e, canvas);
  }

  processMouseReleased(model: DebuggerModel): boolean {
    return model.getCameraController().processPointerUp();
  }

  processMouseDragged(model: DebuggerModel, e: PointerEvent, canvas: HTMLCanvasElement): boolean {
    return model.getCameraController().processPointerMove(e, canvas);
  }

  processMouseWheelMoved(model: DebuggerModel, e: WheelEvent, canvas: HTMLCanvasElement): boolean {
    console.log('.');
    return model.getCameraController().processWheel(e, canvas);
  }
}
